cdap-api-spark/src/main/java/co/cask/cdap/api/spark/JavaSparkExecutionContext.java

```diff
@@ -453,4 +465,16 @@
    * @throws DatasetInstantiationException if the Dataset doesn't exist
    */
   public abstract <K, V> void saveAsDataset(JavaPairRDD<K, V> rdd, String datasetName, Map<String, String> arguments);
+
+  /**
+   * Saves the given {@link JavaPairRDD} to the given {@link Dataset} with the given set of Dataset arguments.
+   *
+   * @param rdd the {@link JavaPairRDD} to be saved
+   * @param namespace the namespace for the Dataset
+   * @param datasetName name of the Dataset
+   * @param arguments arguments for the Dataset
+   * @throws DatasetInstantiationException if the Dataset doesn't exist
```

Comment:
what are the semantics of this method? If the dataset doesn't exist, you should be able to create a new one, no?